test.c

```diff
@@ -3475,24 +3475,75 @@ test_refresh_no_slash_callback (const char *path)
 {
     _cb_count++;
     CU_ASSERT (*(path + strlen (path) - 1) != '/');
-    return 0;
+    return _cb_timeout;
 }
 
 void
 test_refresh_no_slash ()
 {
-    const char *path = TEST_PATH"/interfaces/eth0/state";
+    const char *path = TEST_PATH"/interfaces/eth0";
 
     _cb_count = 0;
     CU_ASSERT (apteryx_refresh (path, test_refresh_no_slash_callback));
-    
+
     CU_ASSERT (apteryx_get (path) == NULL);
+    usleep(_cb_timeout * 2);
     CU_ASSERT (apteryx_get_tree (TEST_PATH"/interfaces/eth0") == NULL);
     CU_ASSERT (_cb_count == 2);
```

Comment:
Actually, surely this test assertion has just changed. It should only be hit once and there should be not sleep above?

Reply:
Done